### DISCUSSION TITLE
LIS Error while using MERRA2 Forcing

### DISCUSSION THREAD

**Discussion by Realone24** [NASA-LIS/LISF#1095] [Q&A]
Dear all,

I was trying to assimilate soil moisture in LIS using MERRA2 forcing. However it was not able to read the forcing files.
Please let me know how to overcome with this error.  I am getting following error message, I have also attached the lis.conf and lislog files here.

 [INFO] Reading MERRA-2 file (bookend,           1  ... 
 ./INPUT/MERRA2/MERRA2_400/Y2017/M04/MERRA2_400.tavg1_2d_slv_Nx.20170402.nc4
 [ERR] nf90_get_var failed for ps in read_merra2 Stopping.
 [ERR] endrun is being calle
[lis.config.txt](https://github.com/NASA-LIS/LISF/files/8685304/lis.config.txt)
[lislog.0000.txt](https://github.com/NASA-LIS/LISF/files/8685309/lislog.0000.txt)

**Comment by bmcandr** [comment]
Hi @Realone24,

Based on your log the file is being opened without error, but can't seem to find the PS variable. Can you confirm that the file contains the PS variable? Here is an example:

```
% ncdump -h INPUT/MERRA2/MERRA2_400/Y2017/M04/MERRA2_400.tavg1_2d_slv_Nx.20170402.nc4

netcdf MERRA2_400.tavg1_2d_slv_Nx.20170402 {
dimensions:
	lon = 576 ;
	lat = 361 ;
	time = UNLIMITED ; // (24 currently)
variables:
	double lon(lon) ;
		lon:long_name = "longitude" ;
		lon:units = "degrees_east" ;
		lon:vmax = 1.e+15f ;
		lon:vmin = -1.e+15f ;
		lon:valid_range = -1.e+15f, 1.e+15f ;
	double lat(lat) ;
		lat:long_name = "latitude" ;
		lat:units = "degrees_north" ;
		lat:vmax = 1.e+15f ;
		lat:vmin = -1.e+15f ;
		lat:valid_range = -1.e+15f, 1.e+15f ;
...
	float PS(time, lat, lon) ;
		PS:long_name = "surface_pressure" ;
		PS:units = "Pa" ;
		PS:_FillValue = 1.e+15f ;
		PS:missing_value = 1.e+15f ;
		PS:fmissing_value = 1.e+15f ;
		PS:scale_factor = 1.f ;
		PS:add_offset = 0.f ;
		PS:standard_name = "surface_pressure" ;
		PS:vmax = 1.e+15f ;
		PS:vmin = -1.e+15f ;
		PS:valid_range = -1.e+15f, 1.e+15f ;
...
```

**Reply by Realone24** [reply]
Hi @bmcandr ,

Thanks for the reply. I have checked the file it has PS variable. I have attached the file for your reference. 

    float PS(time, lat, lon) ;
                PS:standard_name = "surface_pressure" ;
                PS:long_name = "surface_pressure" ;
                PS:units = "Pa" ;
                PS:_FillValue = 1.e+15f ;
                PS:missing_value = 1.e+15f ;
                PS:fmissing_value = 1.e+15f ;
                PS:vmax = 1.e+15f ;
                PS:vmin = -1.e+15f ;
                PS:valid_range = -1.e+15f, 1.e+15f ;

[MERRA2_400.tavg1_2d_slv_Nx.20170401.nc4.txt](https://github.com/NASA-LIS/LISF/files/8705586/MERRA2_400.tavg1_2d_slv_Nx.20170401.nc4.txt)
